src/model/weights_converter.py

```diff
@@ -0,0 +1,234 @@
+        raise NotImplementedError
+
+
+class PytorchToTensorflowHandlers:
+    @staticmethod
+    def _preprocess_tensorflow(tensorflow_weight):
+        return tensorflow_weight, tensorflow_weight.name, tuple(tensorflow_weight.shape)
+
+    @staticmethod
+    def _preprocess_pytorch(pytorch_weight):
+        return pytorch_weight[1], pytorch_weight[0], tuple(pytorch_weight[1].shape)
+
+    class SameShapeHandler(WeightHandler):
+        """Assigns source_weight to target_weight if the weights have same shape."""
+
+        def _process_weights(self, source_weight, target_weight) -> WeightHandlerReturn:
+            weight_tf, name_tf, shape_tf = PytorchToTensorflowHandlers._preprocess_tensorflow(target_weight)
+            weight_py, name_py, shape_py = PytorchToTensorflowHandlers._preprocess_pytorch(source_weight)
+
+            if shape_tf == shape_py:
+                weight_tf.assign(weight_py.numpy())
+                if self.logger:
+                    self.logger.info(f'{name_tf} was assigned successfully from {name_py}')
+                return WeightHandlerReturn(processed=True, matched_source=True, matched_target=True)
+
+            return WeightHandlerReturn(processed=False, matched_source=False, matched_target=False)
+
+    class ConvolutionHandler(WeightHandler):
+        """Assigns source_weight to target_weight if the weights are 2D convolution weights."""
+
+        def _process_weights(self, source_weight, target_weight) -> WeightHandlerReturn:
+            weight_tf, name_tf, shape_tf = PytorchToTensorflowHandlers._preprocess_tensorflow(target_weight)
+            weight_py, name_py, shape_py = PytorchToTensorflowHandlers._preprocess_pytorch(source_weight)
+
+            if len(shape_tf) == len(shape_py) == 4 and shape_tf == (shape_py[2], shape_py[3], shape_py[1], shape_py[0]):
+                weight_tf.assign(weight_py.numpy().transpose(2, 3, 1, 0))
+                if self.logger:
+                    self.logger.info(f'{name_tf} was assigned successfully from {name_py}, transposed')
+                return WeightHandlerReturn(processed=True, matched_source=True, matched_target=True)
+
+            return WeightHandlerReturn(processed=False, matched_source=False, matched_target=False)
+
+    class DepthwiseTransposedConvolutionHandler(WeightHandler):
+        """
+        Assigns source_weight to target_weight if the source_weight is a depthwise transposed convolution and
+        target_weight is a regular transposed convolution.
+        """
+
+        def _process_weights(self, source_weight, target_weight) -> WeightHandlerReturn:
+            weight_tf, name_tf, shape_tf = PytorchToTensorflowHandlers._preprocess_tensorflow(target_weight)
+            weight_py, name_py, shape_py = PytorchToTensorflowHandlers._preprocess_pytorch(source_weight)
+
+            if 'conv2d_transpose' in name_tf and shape_tf[:2] == shape_py[2:] and shape_tf[3] == shape_py[0] and \
```

Comment:
Perhaps save the result of this evaluations in some variables to improve legibility and self-document

Reply:
sure